### PR TITLE
feat: add caller on relayer

### PR DESCRIPTION
**Pull request by lbertenasco**

### CODE REVIEW

File: contracts/StealthRelayer.sol [hunk 1]
```diff
@@ -27,12 +28,26 @@ contract StealthRelayer is Governable, CollectableDust, StealthTx, IStealthRelay
...
     bytes32 _stealthHash,
     uint256 _blockNumber
-  ) external payable override onlyValidJob(_job) validateStealthTxAndBlock(_stealthHash, _blockNumber) returns (bytes memory _returnData) {
+  )
+    external
+    payable
+    override
+    onlyValidJob(_job)
+    validateStealthTxAndBlock(_stealthHash, _blockNumber)
+    setsCaller()
```
Comment (alejoamiras) [inline contracts/StealthRelayer.sol:48]: I think `setsCaller` needs to be the first one to be called (before all the `validate...()`, right ? If we want to use `caller()` from outside to do any validations.

**Review by alejoamiras — APPROVED** [review]
I don't fully understand what will this add (yet) but I think this is part of the process to validate EOA, after that nasty EIP, right ?

File: contracts/StealthRelayer.sol (hunk 1 above)
Comment (lbertenasco) [reply contracts/StealthRelayer.sol:48]: no, caller is only used by the `_job` as a way to know who is executing the tx.
in our case: `caller` is the `keeper`

**Comment by lbertenasco** [comment]
> I don't fully understand what will this add (yet) but I think this is part of the process to validate EOA, after that nasty EIP, right ?

not really, it was a missing piece on the design.

**Pull request merged**